runtime/verutil/cfrerr.c

```diff
@@ -78,6 +78,44 @@ getJ9CfrErrorBsmMessage(J9PortLibrary* portLib, J9CfrError* error, const U_8* cl
 	return errorString;
 }
 
+const char*
+getJ9CfrErrorMajorVersionMessage(J9PortLibrary* portLib, J9CfrError* error, const U_8* className, UDATA classNameLength)
+{
+	PORT_ACCESS_FROM_PORT(portLib);
+	UDATA allocSize = 0;
+	char *errorString = NULL;
+
+	const char *template = j9nls_lookup_message(J9NLS_ERROR | J9NLS_DO_NOT_APPEND_NEWLINE, J9NLS_CFR_ERR_MAJOR_VERSION2, NULL);
+
+	allocSize = strlen(template) + classNameLength + (MAX_INT_SIZE * 4) + 1;
+	errorString = j9mem_allocate_memory(allocSize, OMRMEM_CATEGORY_VM);
+	if (NULL != errorString) {
+		j9str_printf(PORTLIB, errorString, allocSize, template,
+			classNameLength, className, error->errorMajorVersion, error->errorMinorVersion, error->errorMaxAllowedVersion, error->errorOffset);
```

Comment:
The parameters are out of order since the message was updated.

Reply:
I kept the parameter field while changing the message so it should be fine but I could re-order the numbers so it will make more sense.

Reply:
I don't think it works that way, the parameter needs to be in the order of the format specifiers in the message.

Reply:
Which reminds me, you need to add sample_input fields. Look in the .nls file for examples.

Reply:
> I don't think it works that way, the parameter needs to be in the order of the format specifiers in the message.

I think that's what those numbers are meant for. For example: `printf("%2$d %1$d", 1, 2);` prints `2 1`. But I will change it anyway so it won't confuse other people.

Reply:
I'm not sure. You've have to try it with the str_printf(), which isn't the same as printf(). Maybe you did already. I thought I had trouble with it before. Anyway I agree it's better to have the same order to avoid confusion.